sdk/data/azcosmos/cosmos_location_cache.go

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"net/url"
+	"sync"
+	"time"
+)
+
+const defaultExpirationTime time.Duration = time.Minute * 5
+
+const (
+	none opType = iota
+	read
+	write
+	all
+)
+
+type opType int
+
+type locationUnavailabilityInfo struct {
+	lastCheckTime  time.Time
+	unavailableOps opType
+}
+
+type dbAcctLocationsInfo struct {
+	prefLocations              []string
+	availWriteLocations        []string
+	availReadLocations         []string
+	availWriteEndptsByLocation map[string]url.URL
+	availReadEndptsByLocation  map[string]url.URL
+	writeEndpts                []url.URL
+	readEndpts                 []url.URL
+}
+
+// AcctRegion represents a Azure Cosmos DB Database Account in a specific region.
+type AcctRegion struct {
+	// The name of the database account location in the Azure Cosmos DB service.
+	name string
+	// The URL of the database account location in the Azure Cosmos DB service.
+	endpoint string
+	// props    map[string]interface{}
+}
+
+// AcctProperties represents a container for Azure Cosmos DB databases.
+type AcctProperties struct {
+	// The list of readable regions of this database account from the Azure Cosmos DB service.
+	readRegions []AcctRegion
```

Comment:
Are these meant to be deserialized using JSON? Then are we missing the json decorators?

Reply:
If the type is not public and not exported, do we need to still provide the json tags? I am getting a compiler warning about this.

Reply:
The type needs to be internal, the property public: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/azcosmos/cosmos_offers.go#L18-L20